File: array_problems/findMedian/findMedian.go
package main
import (
    "fmt"
    "sort"
    "strconv"
    )
type ArrayProblems struct{}

func (ap *ArrayProblems) findMedian(arr1 []int, arr2 []int) string {
    merged := append(arr1, arr2...)
    sort.Ints(merged)
    midIndex := len(merged) / 2
    if len(merged) % 2 == 0 {
        return strconv.Itoa((merged[midIndex - 1] + merged[midIndex]) / 2)
    } else {
        return strconv.Itoa(merged[midIndex])
    }
}
func main() {
    array_problems := &ArrayProblems{}
    arr1 := []int{1,2,3,4,5}
    arr2 := []int{4,5,6,7,8}
    median := array_problems.findMedian(arr1, arr2)
    fmt.Println(median)
}

File: array_problems/findMedian/findMedian.cs
using System;
using System.Threading.Tasks;
using System.Collections.Generic;
using System.Linq;

class ArrayProblems {
    public async Task<double> findMedian(List<int> arr1, List<int> arr2) {
        List<int> merged = arr1.Concat(arr2).ToList();
        merged.Sort();
        int midIndex = merged.Count / 2;
        return merged.Count % 2 == 0 ? (merged[midIndex - 1] + merged[midIndex]) / 2.0 : merged[midIndex];
    }
}

public class Program{
    public static async Task Main(string[] args) {
        ArrayProblems array_problems = new ArrayProblems();
        List<int> arr1 = new List<int>{1,2,3,4,5};
        List<int> arr2 = new List<int>{4,5,6,7,8};
        double median = await array_problems.findMedian(arr1, arr2);
        Console.WriteLine(median);
    }
}